### PR TITLE
MTP-2397 text Link component, rel external styling, stories

### PR DESCRIPTION
**Pull request by petterkoding**

### CODE REVIEW

File: mt-kit/core/css/src/global/_links.scss [hunk 1]
```diff
@@ -7,16 +7,40 @@
...
+  }
+}
+
 .mt-link {
   @include focus-outline();
   @include disabled;
   $transport-card-list-font-size: $font-size-label;
 
-  color: var(--color-links);
+  color: var(--color-text);
```
Comment (amundlindstol) [inline mt-kit/core/css/src/global/_links.scss:28]: Keep color-links and change color

File: mt-kit/core/css/src/global/_links.scss [hunk 2]
```diff
@@ -7,16 +7,40 @@
 $transportlink-default-color: 'primary' !default;
 $transportlink-colors: 'error' !default;
 
+p:has(a.is-external) .mt-link.is-external {
```
Comment (amundlindstol) [inline mt-kit/core/css/src/global/_links.scss:10]: move to typography

File: mt-kit/core/css/src/_theme.scss [hunk 3]
```diff
@@ -29,6 +30,7 @@ $breakpoint-xx-large: 1200px;
 $breakpoint-xxx-large: 1600px;
 
 $color-links: 'var(--color-primary)' !default;
+$link-underline-color: var(--color-mt-havdyp);
```
Comment (lha047) [inline mt-kit/core/css/src/_theme.scss:33]: Try to stick to the naming patterns. Would call it color-link-underline

File: mt-kit/core/svelte/src/stories/Links.stories.svelte [hunk 4]
```diff
@@ -403,4 +420,7 @@
     flex-direction: column;
     gap: 1rem;
   }
+  .multi-line-paragraph {
+    max-width: 680px;
```
Comment (lha047) [inline mt-kit/core/svelte/src/stories/Links.stories.svelte:424]: use rem

**Review by lha047 — CHANGES_REQUESTED** [review]
I dont think this design has good enough accessibility. Will discuss with designers.
Setting request changes for now

File: mt-kit/core/css/src/global/_links.scss [hunk 5]
```diff
@@ -58,19 +70,33 @@ $transportlink-colors: 'error' !default;
...
-    --icon-width: 15px;
-    --indent: calc(var(--icon-width) + var(--spacer-xxx-small));
+  a[href$='.pdf'],
+  &.pdf {
     display: inline-block;
-    align-items: center;
-    color: var(--color-links);
-    margin-left: var(--indent);
+    &::before {
+      content: var(--icon-pdf);
```
Comment (lha047) [inline mt-kit/core/css/src/global/_links.scss:77]: Use mixin

File: mt-kit/core/css/src/global/_links.scss [hunk 6]
```diff
@@ -58,19 +70,33 @@ $transportlink-colors: 'error' !default;
...
+      margin-right: var(--spacer-xxx-small);
+      display: inline-block;
+      vertical-align: middle;
+    }
+  }
+
+  &.document {
+    display: inline-flex;
     &::before {
       content: var(--icon-document);
```
Comment (lha047) [inline mt-kit/core/css/src/global/_links.scss:87]: use mixin

File: mt-kit/core/css/src/global/_links.scss [hunk 7]
```diff
@@ -58,19 +70,33 @@ $transportlink-colors: 'error' !default;
...
-    --indent: calc(var(--icon-width) + var(--spacer-xxx-small));
+  a[href$='.pdf'],
+  &.pdf {
     display: inline-block;
-    align-items: center;
-    color: var(--color-links);
-    margin-left: var(--indent);
+    &::before {
+      content: var(--icon-pdf);
+      margin-right: var(--spacer-xxx-small);
```
Comment (petterkoding) [inline mt-kit/core/css/src/global/_links.scss:78]: use mixin

File: mt-kit/core/css/src/global/_links.scss [hunk 8]
```diff
@@ -58,19 +70,33 @@ $transportlink-colors: 'error' !default;
...
       display: inline-block;
-      width: var(--icon-width);
-      margin-left: calc(-1 * var(--indent));
+    }
+  }
+  &.back-arrow {
+    display: flex;
+    width: fit-content;
+    &::before {
+      content: var(--icon-carret-left);
```
Comment (lha047) [inline mt-kit/core/css/src/global/_links.scss:96]: use mixin

File: mt-kit/core/svelte/src/lib/ts/utils.ts [hunk 9]
```diff
@@ -122,6 +122,14 @@ export function mapRelExternal(url?: string): 'external' | undefined {
   return url?.startsWith('http') ? 'external' : undefined
 }
 
+export function isExternalLink(href: string): boolean {
```
Comment (lha047) [inline mt-kit/core/svelte/src/lib/ts/utils.ts:125]: remove?

File: mt-kit/core/css/src/global/_links.scss [hunk 10]
```diff
@@ -26,7 +38,7 @@ $transportlink-colors: 'error' !default;
   }
 
   &:hover {
-    text-decoration: none;
+    text-decoration: $color-link-underline underline $border-width-medium;
```
Comment (petterkoding) [inline mt-kit/core/css/src/global/_links.scss:41]: try shorthand

**Review by lha047 — CHANGES_REQUESTED** [review]

File: mt-kit/core/svelte/src/lib/ts/utils.ts (hunk 9 above)
Comment (petterkoding) [reply mt-kit/core/svelte/src/lib/ts/utils.ts]: removed because its not in use

File: mt-kit/core/css/src/global/_links.scss (hunk 8 above)
Comment (petterkoding) [reply mt-kit/core/css/src/global/_links.scss]: fixed

File: mt-kit/core/css/src/global/_links.scss (hunk 6 above)
Comment (petterkoding) [reply mt-kit/core/css/src/global/_links.scss]: fixed

File: mt-kit/core/css/src/global/_links.scss (hunk 5 above)
Comment (petterkoding) [reply mt-kit/core/css/src/global/_links.scss]: fixed

File: mt-kit/core/css/src/global/_links.scss (hunk 10 above)
Comment (petterkoding) [reply mt-kit/core/css/src/global/_links.scss]: `text-decoration-thickness` worked

**Review by lha047 — APPROVED** [review]
Nice :D

**Pull request merged**